src/vs/workbench/contrib/notebook/browser/contrib/multicursor/notebookMulticursor.ts

```diff
@@ -898,14 +1050,53 @@ export class NotebookMultiCursorController extends Disposable implements INotebo
 		this.anchorDisposables.dispose();
 		this.cursorsDisposables.dispose();
 
-		this.trackedMatches.forEach(match => {
-			this.clearDecorations(match);
+		this.trackedCells.forEach(cell => {
+			this.clearDecorations(cell);
 		});
-		this.trackedMatches = [];
+		this.trackedCells = [];
 	}
 
 }
 
+class NotebookSelectAllFindMatches extends NotebookAction {
+	constructor() {
+		super({
+			id: NOTEBOOK_SELECT_ALL_FIND_MATCHES_ID,
+			title: localize('selectAllFindMatches', "Select All Find Matches"),
+			precondition: ContextKeyExpr.and(
+				ContextKeyExpr.equals('config.notebook.multiCursor.enabled', true),
+				NOTEBOOK_IS_ACTIVE_EDITOR,
+				NOTEBOOK_CELL_EDITOR_FOCUSED,
+			),
+			keybinding: {
+				when: ContextKeyExpr.and(
+					ContextKeyExpr.equals('config.notebook.multiCursor.enabled', true),
+					NOTEBOOK_IS_ACTIVE_EDITOR,
+					NOTEBOOK_CELL_EDITOR_FOCUSED,
+				),
+				primary: KeyMod.CtrlCmd | KeyMod.Shift | KeyCode.KeyL,
+				weight: KeybindingWeight.WorkbenchContrib
+			}
+		});
+	}
+
+	override async runWithContext(accessor: ServicesAccessor, context: INotebookActionContext): Promise<void> {
+		const editorService = accessor.get(IEditorService);
+		const editor = getNotebookEditorFromEditorPane(editorService.activeEditorPane);
+
+		if (!editor) {
+			return;
+		}
+
+		if (!context.cell) {
+			return;
+		}
+
+		const controller = editor.getContribution<NotebookMultiCursorController>(NotebookMultiCursorController.id);
```

Comment:
When we have the find widget open, we can use the find controller/model to access the matches instead of searching again.